src/mixings/logit_sb_mixing.cc

```diff
@@ -0,0 +1,164 @@
+#include "logit_sb_mixing.h"
+
+#include <google/protobuf/stubs/casts.h>
+
+#include <Eigen/Dense>
+#include <numeric>
+#include <stan/math/prim.hpp>
+#include <vector>
+
+#include "mixing_prior.pb.h"
+#include "mixing_state.pb.h"
+#include "src/utils/proto_utils.h"
+#include "src/utils/rng.h"
+
+void LogitSBMixing::initialize() {
+  if (prior == nullptr) {
+    throw std::invalid_argument("Mixing prior was not provided");
+  }
+  auto priorcast = cast_prior();
+  num_components = priorcast->num_components();
+  initialize_state();
+  acceptance_rates = Eigen::VectorXd::Zero(num_components - 1);
+  n_iter = 0;
+}
+
+void LogitSBMixing::initialize_state() {
+  auto priorcast = cast_prior();
+  if (priorcast->has_normal_prior()) {
+    Eigen::VectorXd prior_vec =
+        bayesmix::to_eigen(priorcast->normal_prior().mean());
+    dim = prior_vec.size();
+    state.precision = stan::math::inverse_spd(
+        bayesmix::to_eigen(priorcast->normal_prior().var()));
+    if (dim != state.precision.cols()) {
+      throw std::invalid_argument(
+          "Hyperparameters dimensions are not consisent");
+    }
+    if (priorcast->step_size() <= 0) {
+      throw std::invalid_argument("Step size parameter must be > 0");
+    }
+
+    state.regression_coeffs = Eigen::MatrixXd(dim, num_components - 1);
+    for (int i = 0; i < num_components - 1; i++) {
+      state.regression_coeffs.col(i) = prior_vec;
+    }
+
+  } else {
+    throw std::invalid_argument("Unrecognized mixing prior");
+  }
+}
+
+Eigen::VectorXd LogitSBMixing::grad_log_full_cond(
+    const Eigen::VectorXd &alpha, const unsigned int clust,
+    const std::vector<unsigned int> &allocations) {
+  auto priorcast = cast_prior();
+  Eigen::VectorXd prior_mean =
+      bayesmix::to_eigen(priorcast->normal_prior().mean());
+  Eigen::VectorXd grad = state.precision * (prior_mean - alpha);
+  for (int i = 0; i < allocations.size(); i++) {
+    if (allocations[i] >= clust) {
+      bool is_curr_clus = (allocations[i] == clust);
+      double prob = sigmoid(covariates_ptr->row(i).dot(alpha));
+      grad += (is_curr_clus - prob) * covariates_ptr->row(i);
+    }
+  }
+  return grad;
+}
+
+double LogitSBMixing::log_like(const Eigen::VectorXd &alpha,
```

Comment:
is this really a `log_like`? I would just have a `full_cond_lpdf` that returns what this is returning + the prior evaluation 

The A/R ratio becomes `full_cond_ratio - prop_ratio` where `full_cond_ratio = full_cond_lpdf(state_prop, h, allocations) - full_cond_lpdf(state_c, h, allocations)`

Reply:
Good idea, I'll change it, we don't really need a separate likelihood member.
Speaking of which, I think this is indeed a likelihood if you treat the nu_ih as data. Which they kind of are, if you think of their relation to alpha as a generalized linear model. Is it? (I'm asking just for theoretical understanding)

Reply:
"Everything is a likelihood if you interpret the parameters hard enough" (M. Ghandi)